src/main/java/favourite/use_case/PostStringFactory.java

```diff
@@ -0,0 +1,16 @@
+package favourite.use_case;
+
+import entities.Post;
+
+import java.util.ArrayList;
+
+/**
+ * PostStringFactory in the use case layer implements PostWriterInterface.
+ * It converts Post objects into a format that the database can save.
+ */
+public class PostStringFactory implements PostWriterInterface {
```

Comment:
A useful way to do this is to use a mapping from the kind of attribute (title, poster, etc) to actual data, and pass that through to the dataaccess and have it interpret this

Reply:
Moreover, I don't think we need a 'factory' to do this, as most of the interaction with the data gateway should be done through the interactor. I believe this task could be done as a method in the interactor, which would then pass it to the database, as opposed to having the database call something in the use case layer to do it

Reply:
I think the class name should be changed from facotory to avoid confusion, but it does not need the interface IMO as it should just be called from the interactor, and then have its return argument passed to dataaccess